test/Sema/call_as_function_simple.swift

```diff
@@ -106,8 +106,12 @@ struct Mutating {
   }
 }
 func testMutating(_ x: Mutating, _ y: inout Mutating) {
-  _ = x() // expected-error {{cannot use mutating member on immutable value: 'x' is a 'let' constant}}
-  _ = x.callAsFunction() // expected-error {{cannot use mutating member on immutable value: 'x' is a 'let' constant}}
+  // TODO: Improve this error to match the error using a direct `callAsFunction` member reference.
```

Comment:
Note: this diagnostic regressed: the diagnostic is now produced from failed `member` constraint instead of custom `CSApply` diagnostic hack. cc @xedin regarding ideas for improving/fixing this.

The diagnostic below in `testInout` has improved however - it is now in parity with direct `callAsFunction` calls.